### PR TITLE
🐛 Use unique ids for mondoo spaces. Improve provider code

### PR DESCRIPTION
**Pull request by preslavgerchev**
If we're connecting with an org config, this won't work. Switch to using the space's MRN from the MQL resource.

### CODE REVIEW

File: providers/mondoo/resources/mondoo_space.go [hunk 1]
```diff
@@ -39,7 +57,7 @@ func (m *mqlMondooSpace) assets() ([]any, error) {
 		} `graphql:"assets(spaceMrn: $spaceMrn)"`
 	}
 	vars := map[string]any{
-		"spaceMrn": mondoogql.String(conn.Upstream.SpaceMrn),
+		"spaceMrn": mondoogql.String(m.Mrn.Data),
```
Comment (preslavgerchev) [inline providers/mondoo/resources/mondoo_space.go:60]: this means we use the mql MRN to fetch data. this means if we connect via a mondoo org this still works

File: providers/mondoo/resources/mondoo_organization.go [hunk 2]
```diff
@@ -5,12 +5,29 @@ package resources
...
 	mondoogql "go.mondoo.com/mondoo-go"
 )
 
+func initMondooOrganization(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
+	conn := runtime.Connection.(*connection.Connection)
+	if conn.Type != connection.ConnTypeOrganization {
+		return nil, nil, errors.New("cannot initialize mondoo.organization, invalid connection")
+	}
+	args["mrn"] = llx.StringData(conn.Upstream.SpaceMrn)
+	args["name"] = llx.StringData(connection.MrnBasenameOrMrn(conn.Upstream.SpaceMrn))
```
Comment (chris-rock) [inline providers/mondoo/resources/mondoo_organization.go:22]: That is the space id, right?

File: providers/mondoo/provider/provider.go [hunk 3]
```diff
@@ -145,55 +124,7 @@ func fillAsset(mrnStr string, asset *inventory.Asset) error {
...
-	_, err = resources.CreateResource(runtime, "mondoo.client", map[string]*llx.RawData{
-		"mrn": llx.StringData(conn.Upstream.Creds.Mrn),
-	})
-	if err != nil {
-		return err
-	}
-
-	spaceMrn := conn.Upstream.SpaceMrn
-	if asset.Platform.Name == "mondoo-space" {
-		_, err = resources.CreateResource(runtime, "mondoo.space", map[string]*llx.RawData{
```
Comment (preslavgerchev) [inline providers/mondoo/provider/provider.go:179]: this stopped working once i added `id()` funcs to mondoo.space and mondoo.organization

i didnt dive into why, but i did move this to be init funcs for the mql resources

File: providers/mondoo/resources/mondoo_space.go [hunk 4]
```diff
@@ -5,14 +5,32 @@ package resources
...
 )
 
+func initMondooSpace(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
+	conn := runtime.Connection.(*connection.Connection)
+	if conn.Type != connection.ConnTypeSpace {
+		return nil, nil, errors.New("cannot initialize mondoo.space, invalid connection")
+	}
+	args["mrn"] = llx.StringData(conn.Upstream.SpaceMrn)
+	// TODO: we should fetch this from the API, mrn basename != name
+	args["name"] = llx.StringData(connection.MrnBasenameOrMrn(conn.Upstream.SpaceMrn))
```
Comment (chris-rock) [inline providers/mondoo/resources/mondoo_space.go:25]: I think we should turn that into a resolver to fetch the space details.

File: providers/mondoo/resources/mondoo_organization.go (hunk 2 above)
Comment (preslavgerchev) [reply providers/mondoo/resources/mondoo_organization.go:22]: this is either the org id or the space id, regardless of the fact we call it `SpaceMrn`, it all depends on what the mondoo config that we're connecting with has

File: providers/mondoo/resources/mondoo_space.go (hunk 4 above)
Comment (preslavgerchev) [reply providers/mondoo/resources/mondoo_space.go:25]: +1, thats the TODO above that line

lets do it in a follow-up PR

**Review by chris-rock — APPROVED** [review]
Thank you @preslavgerchev

**Pull request merged**